### PR TITLE
Added info about using merged-kubeconfig for joining the cluster.

### PR DESCRIPTION
**Pull request by aswinsuryan**
Signed-off-by: Aswin Surayanarayanan <asuryana@redhat.com>

Fixes: #210

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,15 +66,20 @@ The [Lighthouse](https://github.com/submariner-io/lighthouse) project helps in c
...
 
+* Create a kubeconfig file which has the context of all the clusters. If  KUBECONFIG environment variable set with the path to all clusters, you could use the below command.
+```
+kubectl config view --flatten > kubeconfig_merged
+```
+
 * Deploy Broker with Dataplane - To deploy the Submariner broker along with the data cluster, run using the broker-info.subm generated in the folder from which the previous step was run.
 ```
-subctl deploy-broker --kubeconfig <PATH-TO-KUBECONFIG-BROKER> --dataplane --service-discovery --broker-cluster-context <BROKER-CONTEXT-NAME> --clusterid  <CLUSTER-ID-FOR-TUNNELS>
+subctl deploy-broker --kubecontext <BROKER-CONTEXT-NAME>  --kubeconfig <PATH-TO-KUBECONFIG-BROKER> --dataplane --service-discovery --broker-cluster-context <BROKER-CONTEXT-NAME> --clusterid  <CLUSTER-ID-FOR-TUNNELS>
```
Comment (skitt) [inline README.md]: Does this step require a non-merged configuration?

File: README.md [hunk 2]
```diff
@@ -66,15 +66,20 @@ The [Lighthouse](https://github.com/submariner-io/lighthouse) project helps in c
 
 To deploy Submariner with Lighthouse follow the below steps.
 
+* Create a kubeconfig file which has the context of all the clusters. If  KUBECONFIG environment variable set with the path to all clusters, you could use the below command.
```
Comment (vthapar) [inline README.md]: nit: doublespace between if and KUBECONFG.
nit: "environment variable is set"

File: README.md (hunk 1 above)
Comment (aswinsuryan) [reply README.md]: Yes, else it is failing.

Comment (vthapar) [inline README.md]: Add a warning to highlight that due to a bug deploy-broker requires unmerged kubeconfg and should use workaround of using unmerged config.

> :warning: Due to a bug `deploy-broker` fails if using merged kubeconfig. Use unmerged broker kubeconfig as a workaround while issue is being fixed.

Add link to issue.

File: README.md (hunk 2 above)
Comment (aswinsuryan) [reply README.md]: Done.

File: README.md (hunk 1 above)
Comment (aswinsuryan) [reply README.md]: I think since it is due to Kubefed and removal of kubefed will fix this, it would be ok right if we don't mention an explicit warning. May be general warning in the initial section should be fine.

**Comment by skitt** [comment]
This will also need to be cherry-picked to the 0.1 branch.

File: README.md [hunk 3]
```diff
@@ -66,15 +66,20 @@ The [Lighthouse](https://github.com/submariner-io/lighthouse) project helps in c
 
 To deploy Submariner with Lighthouse follow the below steps.
 
+* Create a kubeconfig file which has the context of all the clusters. If KUBECONFIG environment variable set with the path to all clusters, you could use the below command.
+```
+kubectl config view --flatten > kubeconfig_merged
```
Comment (manosnoam) [inline README.md]: After creating "kubeconfig_merged", where should it be used ?
Please specify command.

File: README.md [hunk 4]
```diff
@@ -66,15 +66,20 @@ The [Lighthouse](https://github.com/submariner-io/lighthouse) project helps in c
...
+```
+
 * Deploy Broker with Dataplane - To deploy the Submariner broker along with the data cluster, run using the broker-info.subm generated in the folder from which the previous step was run.
 ```
-subctl deploy-broker --kubeconfig <PATH-TO-KUBECONFIG-BROKER> --dataplane --service-discovery --broker-cluster-context <BROKER-CONTEXT-NAME> --clusterid  <CLUSTER-ID-FOR-TUNNELS>
+subctl deploy-broker --kubecontext <BROKER-CONTEXT-NAME>  --kubeconfig <PATH-TO-KUBECONFIG-BROKER> --dataplane --service-discovery --broker-cluster-context <BROKER-CONTEXT-NAME> --clusterid  <CLUSTER-ID-FOR-TUNNELS>
```
Comment (manosnoam) [inline README.md]: Before calling deploy command, what if the current kubeconfig (already exported) is pointing to one of the clusters, which is not the broker cluster ?

Comment (aswinsuryan) [reply README.md]: --kubecontext overides the current-cluster selection.

File: README.md (hunk 3 above)
Comment (aswinsuryan) [reply README.md]: I have added the details in the join command, where this should be used.

Comment (aswinsuryan) [reply README.md]: I have added more details on the README based on your feedback.

File: README.md [hunk 5]
```diff
@@ -66,17 +66,32 @@ The [Lighthouse](https://github.com/submariner-io/lighthouse) project helps in c
 
 To deploy Submariner with Lighthouse follow the below steps.
 
-* Deploy Broker with Dataplane - To deploy the Submariner broker along with the data cluster, run using the broker-info.subm generated in the folder from which the previous step was run.
+* Create a kubeconfig file which has the context of all the clusters. If KUBECONFIG environment variable set with the path to all clusters(like KUBECONFIG=/home/user/.config/east/auth/kubeconfig-dev:/home/user/.config/west/auth/kubeconfig-dev), you could use the below command.
```
Comment (manosnoam) [inline README.md]: Thanks for adding all examples, and explanations. 
please just add space in "all clusters (like"

Comment (aswinsuryan) [reply README.md]: Done.

File: README.md [hunk 6]
```diff
@@ -66,17 +66,32 @@ The [Lighthouse](https://github.com/submariner-io/lighthouse) project helps in c
 
 To deploy Submariner with Lighthouse follow the below steps.
 
-* Deploy Broker with Dataplane - To deploy the Submariner broker along with the data cluster, run using the broker-info.subm generated in the folder from which the previous step was run.
+* Create a kubeconfig file which has the context of all the clusters (including the Broker cluster). If KUBECONFIG environment variable set with the path to all clusters (like KUBECONFIG=/home/user/.config/east/auth/kubeconfig-dev:/home/user/.config/west/auth/kubeconfig-dev), you could use the below command.
```
Comment (manosnoam) [inline README.md]: Specifying "If KUBECONFIG environment variable set with...", is not optional, but mandatory. Please add:
export KUBECONFIG="${KUBECONFIG_BROKER}:${KUBECONFIG_A}:${KUBECONFIG_B}"

Comment (skitt) [reply README.md]: No, it’s not mandatory, there are other ways to create a merged file. However, being directive is better in end-user documentation, so we could rephrase this to say “export ...” with no conditional.

File: README.md [hunk 7]
```diff
@@ -43,11 +43,11 @@ If `~/.local/bin` is on your PATH, you will then be able to run subctl.
 
 To deploy submariner in a two Kubernetes cluster setup follow the below steps
 
-* Deploy Broker with Dataplane - To deploy the Submariner broker along with the data cluster run
+* Deploy Broker with Dataplane - To deploy the Submariner broker along with the dataplane on the same cluster run:
 ```
 subctl deploy-broker --kubeconfig <PATH-TO-KUBECONFIG-BROKER> --dataplane --clusterid  <CLUSTER-ID-FOR-TUNNELS>
 ```
-
+Note, if --dataplane (enabled by default) was set, it is not required to join this cluster, as it was already joined. It is only required to join the other clusters.
```
Comment (mangelajo) [inline README.md]:
```suggestion
Note, if --dataplane (disabled by default) was set, it is not required to join this cluster, as it was already joined. It is only required to join the other clusters.
```

File: README.md [hunk 8]
```diff
@@ -66,17 +66,32 @@ The [Lighthouse](https://github.com/submariner-io/lighthouse) project helps in c
 
 To deploy Submariner with Lighthouse follow the below steps.
 
-* Deploy Broker with Dataplane - To deploy the Submariner broker along with the data cluster, run using the broker-info.subm generated in the folder from which the previous step was run.
+* Create a kubeconfig file which has the context of all the clusters (including the Broker cluster). If KUBECONFIG environment variable set with the path to all clusters (like KUBECONFIG=/home/user/.config/east/auth/kubeconfig-dev:/home/user/.config/west/auth/kubeconfig-dev), you could use the below command.
 ```
-subctl deploy-broker --kubeconfig <PATH-TO-KUBECONFIG-BROKER> --dataplane --service-discovery --broker-cluster-context <BROKER-CONTEXT-NAME> --clusterid  <CLUSTER-ID-FOR-TUNNELS>
+kubectl config view --flatten > merged_kubeconfig
```
Comment (mangelajo) [inline README.md]: But this does not work for openshift configs, all clusters will have context "admin".

We need a solution for that as part of #210

**Comment by mangelajo** [comment]
@aswinsuryan @skitt @vthapar see this for reference: https://www.youtube.com/watch?v=zuh3JFtNbeU

Probabably we can include that step as part of an openshift with discovery quickstart

File: README.md (hunk 7 above)
Comment (aswinsuryan) [reply README.md]: Done.

File: README.md (hunk 6 above)
Comment (aswinsuryan) [reply README.md]: Done.

File: README.md [hunk 9]
```diff
@@ -66,17 +66,44 @@ The [Lighthouse](https://github.com/submariner-io/lighthouse) project helps in c
 
 To deploy Submariner with Lighthouse follow the below steps.
 
-* Deploy Broker with Dataplane - To deploy the Submariner broker along with the data cluster, run using the broker-info.subm generated in the folder from which the previous step was run.
+* Create a kubeconfig file which has the context of all the clusters (including the Broker cluster). you could use the below command after exporting the kubeconfigs of all the clusters( eg export KUBECONFIG=/home/user/.config/east/auth/kubeconfig-dev:/home/user/.config/west/auth/kubeconfig-dev)
+```
+kubectl config view --flatten > merged_kubeconfig
+```
+
+Note, if you are using Openshift configs and if it has the default name for users (admin), merging the files will create a conflict. So you can change the name to a meaningful name before running the above command using,
```
Comment (mangelajo) [inline README.md]: :+1:

**Comment by mangelajo** [comment]
> This will also need to be cherry-picked to the 0.1 branch.

Technically I think we don't need to do that since we are not publishing separate docs. I think we may want to move all this to the website instead when we are comfortable about it.

**Comment by skitt** [comment]
> > This will also need to be cherry-picked to the 0.1 branch.
> 
> Technically I think we don't need to do that since we are not publishing separate docs. I think we may want to move all this to the website instead when we are comfortable about it.

Even if we don’t *publish* branched docs anywhere, it’s nice to provide the most relevant docs in the source tarballs from the releases page ;-).

**Comment by skitt** [comment]
Cherry-picked in #233.
